### PR TITLE
Fix up ClusterServiceIT

### PR DESCRIPTION
**Pull request by DaveCTurner**
ClusterServiceIT#testPendingUpdateTask has some unbounded waits, it relies on the clock advancing by at least 1ms which might not happen, and it leaves the cluster service thread blocked on failure which causes knock-on effects. This commit addresses these problems.

### CODE REVIEW

File: server/src/internalClusterTest/java/org/elasticsearch/cluster/service/ClusterServiceIT.java [hunk 1]
```diff
@@ -397,40 +397,48 @@ public void onFailure(Exception e) {
...
 
-                @Override
-                public void onFailure(Exception e) {
-                    fail();
-                }
-            });
-        }
-        Thread.sleep(100);
+            final var startNanoTime = System.nanoTime();
+            while (TimeUnit.MILLISECONDS.convert(System.nanoTime() - startNanoTime, TimeUnit.NANOSECONDS) <= 0) {
```
Comment (kingherc) [inline server/src/internalClusterTest/java/org/elasticsearch/cluster/service/ClusterServiceIT.java:418]: This looks weird to me. Does this mean that you wait at least for 1ms to have passed? And if not, you then wait for 100ms (which defeats the purpose of the while loop -- except for rare cases where the sleep is interrupted within 1ms)?

Comment (kingherc) [reply server/src/internalClusterTest/java/org/elasticsearch/cluster/service/ClusterServiceIT.java:418]: Ah re-reading the PR description seems to clarify this now. Indeed you want to ensure that it waits at least 1ms. Couldn't we better though try-catch that thread.sleep for InterruptedException?

Comment (DaveCTurner) [reply server/src/internalClusterTest/java/org/elasticsearch/cluster/service/ClusterServiceIT.java:418]: I encountered a test failure where even the existing 100ms sleep didn't result in `System.nanoTime()` returning a newer time.

I don't think we need to catch an `InterruptedException` here, we can just fail the test in that case and let the test runner work out what to do with the interrupt.

**Review by kingherc — APPROVED** [review]
LGTM

**Comment by DaveCTurner** [comment]
Ah, I just realised the failure I was chasing was on a branch that uses a much coarser clock for these stats than `System::nanoTime`. This is still a good fix, just much less important than I thought.

**Pull request merged**